From deb9d9412528ed22e6a747551bc1f729e9d5b3c1 Mon Sep 17 00:00:00 2001
From: Li Jun <b47624@freescale.com>
Date: Thu, 16 Oct 2014 22:17:52 +0800
Subject: [PATCH] MLK-9617-15 usb: gadget: composite: add get otg descriptor
 request handling

This patch adds response to getting OTG descriptor request from host.

Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit 560690f1b24448b94806f390add9d91d602e3ef4)
---
 drivers/usb/gadget/composite.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/usb/gadget/composite.c b/drivers/usb/gadget/composite.c
index 66de2a0..9cf244f 100644
--- a/drivers/usb/gadget/composite.c
+++ b/drivers/usb/gadget/composite.c
@@ -1286,6 +1286,19 @@ composite_setup(struct usb_gadget *gadget, const struct usb_ctrlrequest *ctrl)
 			if (value >= 0)
 				value = min(w_length, (u16) value);
 			break;
+		case USB_DT_OTG:
+			if (gadget_is_otg(gadget)) {
+				struct usb_configuration *config;
+
+				config = list_first_entry(&cdev->configs,
+						struct usb_configuration, list);
+				if (!config)
+					goto done;
+
+				value = sizeof(struct usb_otg_descriptor);
+				memcpy(req->buf, config->descriptors[0], value);
+			}
+			break;
 		case USB_DT_BOS:
 			if (gadget_is_superspeed(gadget)) {
 				value = bos_desc(cdev);
-- 
1.8.0

